util/json.h

```diff
@@ -16,14 +16,14 @@
 	json_object_object_add(o, k, json_object_new_int(v))
 #ifdef CONFIG_JSONC_14
 #define json_object_add_value_uint64(o, k, v) \
-	json_object_object_add(o, k, json_object_new_uint64(v))
+	json_object_object_add(o, k, json_object_new_int64(v))
 #else
 #define json_object_add_value_uint64(o, k, v) \
-	if ((v) > UINT_MAX) {						\
-		fprintf(stderr, "Value overflow in %s", k);		\
-		json_object_object_add(o, k, json_object_new_int(-1));	\
-	} else								\
-		json_object_object_add(o, k, json_object_new_int(v))
+	if ((v) > ULONG_MAX) { \
+		fprintf(stderr, "\nValue overflow in %s, v = %ld\n", k, (long)v); \
+		json_object_object_add(o, k, json_object_new_int64(-1)); \
```

Comment:
And did you check if this code actually compiles against json-c v0.13? The very reason we have this workaround here is because older version of json-c do not have this API.

3c3b5920f77c ("add uint64 data to json-c")

```
$ git tag --contains 3c3b5920f77c751471cd46b92f9d3f5aa6ff2557                                                                                                                json-c-0.14-20200419
json-c-0.15-20200726
json-c-0.16-20220414
```